presto-main/src/main/java/io/prestosql/sql/planner/Partitioning.java

```diff
@@ -59,11 +61,9 @@ public static Partitioning create(PartitioningHandle handle, List<Symbol> column
                 .collect(toImmutableList()));
     }
 
-    public static Partitioning createWithExpressions(PartitioningHandle handle, List<Expression> expressions)
+    public static Partitioning createWithBindings(PartitioningHandle handle, List<ArgumentBinding> bindings)
     {
-        return new Partitioning(handle, expressions.stream()
-                .map(ArgumentBinding::expressionBinding)
-                .collect(toImmutableList()));
+        return new Partitioning(handle, bindings);
```

Comment:
It would be nice to add some javadoc what `List<ArgumentBinding> arguments` represents.

Reply:
Yes. I can do that as a separate PR. In a nutshell, `Partitioning` represents as fully formed "function call" of some sort. `List<ArgumentBinding>` is just list of arguments being passed to that function. They can be either an expression or a constant, but since we have no way of representing arbitrary constants in the IR, we use a union structure. Once we decouple of AST from the IR, we'll be able to replace `Partitioning` with the equivalent of a `FunctionCall`.